modules/channel-slack/src/backend/client.ts

```diff
@@ -0,0 +1,164 @@
+import { createMessageAdapter } from '@slack/interactive-messages'
+import { RTMClient } from '@slack/rtm-api'
+import { WebClient } from '@slack/web-api'
+import axios from 'axios'
+import * as sdk from 'botpress/sdk'
+import { Router } from 'express'
+import _ from 'lodash'
+
+import { Config } from '../config'
+
+import { Clients } from './typings'
+
+const debug = DEBUG('channel-slack')
+const debugIncoming = debug.sub('incoming')
+const debugOutgoing = debug.sub('outgoing')
+
+const outgoingTypes = ['text', 'image', 'actions', 'typing', 'carousel']
+
+export class SlackClient {
+  private router: Router & sdk.http.RouterExtension
+  private client: WebClient
+  private rtm: RTMClient
+  private interactive: any
+  private botId: string
+
+  constructor(private bp: typeof sdk, botId: string, config: Config, router) {
+    this.botId = botId
+    this.router = router
+
+    this.client = new WebClient(config.botToken)
+    this.rtm = new RTMClient(config.botToken)
+    this.interactive = createMessageAdapter(config.signingSecret, {}) as any
```

Comment:
what would you think of `adapter` instead of `interactive`? It fits method name.

Reply:
Nop, because the client is actually for interactive events. there's no typings for that particular one. Adapter doesn't make sense